riptide-metrics is mandatory dependency when using io.dropwizard.metrics:metrics-core in project

Riptide autoconfiguration assumes riptide-metrics is in classpath as soon as there is a `com.codahale.metrics.MetricRegistry` bean configured. This leads to a 
```
Caused by: java.lang.NoClassDefFoundError: org/zalando/riptide/metrics/MetricsNameGenerator
        at java.lang.Class.getDeclaredMethods0(Native Method)
        at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
        at java.lang.Class.getDeclaredMethods(Class.java:1975)
        at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:613)
        ...
```
on application start.

https://github.com/zalando/riptide/blob/5efafaba5fca3e8a2d875602ecac97089541d721/riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/RiptideAutoConfiguration.java#L33

--- a/riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/RiptideAutoConfiguration.java
+++ b/riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/RiptideAutoConfiguration.java
@@ -5,6 +5,7 @@ import lombok.AllArgsConstructor;
 import org.springframework.boot.actuate.metrics.GaugeService;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
@@ -26,6 +27,7 @@ public class RiptideAutoConfiguration {
     }
 
     @Configuration
+    @ConditionalOnClass(MetricsPlugin.class)
     @ConditionalOnMissingBean(MetricsPlugin.class)
     @ConditionalOnBean(MetricRegistry.class)
     static class MetricsConfiguration {

